docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/src/libraries/charts/utilities/index.js

```diff
@@ -57,3 +57,73 @@ export const getShouldLoadChartWrapper = (pathName, rules) => {
 
     return false
 };
+
+/**
+ * buildAxisData - Builds out axis data labels by running against provided formatter
+ *
+ * @param {[]} axisData
+ * @param labelFormatter
+ * @returns {*}
+ */
+export function buildAxisData(axisData, labelFormatter) {
+  for (let i = 0; i < axisData.length; i++) {
+    if (axisData[i].labels.hasFormatter) {
+      delete axisData[i].labels.hasFormatter;
+      axisData[i].labels.formatter = labelFormatter[i].formatter;
+    }
+  }
+  return axisData;
+};
+
+/**
+ * getChartData - Returns chart data based on provided data file name
+ *
+ * @param {String} dataFileName
+ * @returns {*}
+ */
+export const getChartData = (dataFileName) => {
+  const sessionKey = 'factBook';
+
+  // Get chart data from session
+  const chartData = getSessionChartData(sessionKey);
```

Comment:
This seems like a premature optimization.  None of these data files should be particularly large.

Reply:
It's not the size I was concerned about here, but the unnecessary multiple network calls.

Reply:
I understand what you're trying to do, but are those connections actually a problem? Do you know that multiple requests will take place? (This seems to expect that the user will reload the page multiple times.)

This adds complexity without a clear benefit. It's also likely to cause confusion when the data is updated and refreshing the browser doesn't result in a change. (The underlying `window.sessionStorage` object will persist until the tab is closed.)

Reply:
They were a problem, as the way the charts are initiated and loaded caused all chart calls to be made each time any factbook url is visited, which is the way the chart works. However, in the recent push, I made some changes as to how, and when data is injected which no longer necessitates multiple calls thereby removing the need to cache the results. `window.sessionStorage` in no longer in use.

Reply:
Thanks! That seems like an all-around better way to handle it.